From e23f280096410094349d7f2aa2803503c91a3974 Mon Sep 17 00:00:00 2001
From: Mohan Kumar G <quic_mkumarg@quicinc.com>
Date: Mon, 18 Dec 2023 10:00:00 +0530
Subject: [PATCH] hostapd: enable RNR for hidden bss

Currently when a bss link is hidden, it is not advertised
in partner BSS's RNR. So when all links are hidden in MLO,
association happens only in single link.

This is because when all links are hidden, none of the bss advertises
its partner and hence association does not happen in other links.

Fix this issue by removing the check for hidden bss in the functions
hostapd_eid_rnr_iface and hostapd_eid_rnr_iface_len.

Signed-off-by: Mohan Kumar G <quic_mkumarg@quicinc.com>
---
 src/ap/ieee802_11.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -7924,8 +7924,7 @@ repeat_rnr_len:
 			ap_mld = bss->conf->mld_ap;
 #endif /* CONFIG_IEEE80211BE */
 
-			if (bss == reporting_hapd ||
-			    bss->conf->ignore_broadcast_ssid)
+			if (bss == reporting_hapd)
 				continue;
 
 			if (hostapd_skip_rnr(i, skip_profiles, ap_mld,
@@ -8188,7 +8187,7 @@ static bool hostapd_eid_rnr_bss(struct h
 #endif /* CONFIG_IEEE80211BE */
 
 	if (!bss || !bss->conf || !bss->started ||
-	    bss == reporting_hapd || bss->conf->ignore_broadcast_ssid)
+	    bss == reporting_hapd)
 		return false;
 
 	if (hostapd_skip_rnr(i, skip_profiles, ap_mld, tbtt_info_len,
